### PR TITLE
vf_vapoursynth: add display refresh rate property

### PR DESCRIPTION
**Pull request by Argon-**
For me it seems to work quite reliably, at least I couldn't break it. 
`display_fps` is 0 in a VS script when there's not (yet) a refresh rate set by the back-end.

### CODE REVIEW

**Comment by haasn** [comment]
Does this work for mvtools?

**Comment by Argon-** [comment]
I'm successfully using it in `core.mv.BlockFPS`, you just have to split it into `num` and `den` like `container_fps`.
But I guess the usefulness depends on the back-end, i.e. if it's actually delivering a correct/sane value.

**Comment by wm4** [comment]
Can someone test what this does on x11?

**Comment by haasn** [comment]
Added this to my script

```python
print("display_fps", display_fps)
```

Here's my output:

```
Playing: gits_sample.mkv
 (+) Video --vid=1 (h264)
 (+) Audio --aid=1 --alang=jpn (*) (wavpack)
File tags:
 Title: Ghost in the Shell: Stand Alone Complex — Solid State Society
[vo/opengl-hq] Opening ICC profile '~/icc/main.icc'
[vo/opengl-hq] Opening 3D LUT cache in file '~/.cache/3dlut'.
Opening video filter: [format fmt=yuv420p]
Opening video filter: [vapoursynth file=~/.mpv/mvtools.vpy buffered-frames=2 concurrent-frames=8]
AO: [pulse] 48000Hz 5.1(side) 6ch s32
Using conversion filter.
display_fps 0.0
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.
VO: [opengl-hq] 960x540 yuv420p
Resolution: 960x540, Framerate: 23.976 Hz
AV: 00:00:03 / 00:01:37 (3%) A-V:  0.000

[vapoursynth] Filter error at frame 40: EOF or filter reinit/uninit
[vapoursynth] Filter error at frame 41: EOF or filter reinit/uninit
[vapoursynth] Filter error at frame 42: EOF or filter reinit/uninit
[vapoursynth] Filter error at frame 39: EOF or filter reinit/uninit
[vapoursynth] Filter error at frame 38: EOF or filter reinit/uninit
[vapoursynth] Filter error at frame 37: EOF or filter reinit/uninit
[vapoursynth] Filter error at frame 43: EOF or filter reinit/uninit
[vapoursynth] Filter error at frame 44: EOF or filter reinit/uninit
display_fps 0.0
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.

Exiting... (Quit)
```

TL;DR: I get 0.0 both times the filter gets initialized.

Note that the FPS gets detected correctly (the second time around, at least):

```
...
[vo/opengl-hq/x11] Current display FPS: 59.998457
[vo/opengl-hq] Assuming 59.998457 FPS for framedrop.
...
```

**Comment by wm4** [comment]
In theory, you could watch the display fps property in lua (after such a property is added), and then adjust the filters accordingly. But maybe this would be too complicated?

**Comment by haasn** [comment]
Sounds ugly and complicated, and also prone to unnecessarily reinitializing the filter.

(Incidentally, the filter already breaks on movies that have only 1 frame, not sure if that's due to mvtools being bad or due to vf_vapoursynth itself)

**Comment by Argon-** [comment]
Uhm, interesting:

    $ build/mpv ~/Desktop/test1.mkv --vf vapoursynth=~/.mpv/vs-scripts/motion-interpolation.vpy --no-force-window
       cplayer: Playing: /Users/Julian/Desktop/test1.mkv
       cplayer:  (+) Video --vid=1 (h264)
       cplayer:  (+) Audio --aid=1 --alang=jpn (*) 'Japanese 2.0' (aac)
       cplayer:      Audio --aid=2 --alang=eng 'English 2.0' (aac)
       cplayer:  (+) Subs  --sid=1 --slang=eng (*) 'English Subs' (ass)
            vf: Opening video filter: [vapoursynth file=~/.mpv/vs-scripts/motion-interpolation.vpy]
       cplayer: AO: [coreaudio] 48000Hz stereo 2ch float
    display_fps: 60.0
    vapoursynth: Frame requested during init! This is unsupported.
    vapoursynth: Returning black dummy frame with 0 duration.
    vapoursynth: Frame requested during init! This is unsupported.
    vapoursynth: Returning black dummy frame with 0 duration.
       cplayer: VO: [opengl-hq] 1280x720 yuv420p10
    statusline: (Paused) AV: 00:00:00 / 00:00:01 (0%) A-V:  0.000 Cache:  0s+0KB
    [+-------------------------------------------------------------------------------------------------]
       cplayer:
    vapoursynth: Filter error at frame 1: EOF or filter reinit/uninit
    vapoursynth: Filter error at frame 2: EOF or filter reinit/uninit
    vapoursynth: Filter error at frame 3: EOF or filter reinit/uninit
    vapoursynth: Filter error at frame 5: EOF or filter reinit/uninit
    vapoursynth: Filter error at frame 4: EOF or filter reinit/uninit
    vapoursynth: Filter error at frame 6: EOF or filter reinit/uninit
    vapoursynth: Filter error at frame 7: EOF or filter reinit/uninit
    vapoursynth: Filter error at frame 8: EOF or filter reinit/uninit
    display_fps: 60.0
    vapoursynth: Frame requested during init! This is unsupported.
    vapoursynth: Returning black dummy frame with 0 duration.
    vapoursynth: Frame requested during init! This is unsupported.
    vapoursynth: Returning black dummy frame with 0 duration.
       cplayer:
       cplayer: Exiting... (Quit)

I'm not sure why it works for me, I hope this is not a race condition.
Looking at `cocoa_common.m` the refresh rate is not only queried when configuring a VO window but also when getting an ICC profile (I'm using `icc-profile-auto`). I'm not sure if I get `x11_common.c` correctly, but there the refresh rate is queried only when configuring a VO window.
However, for me it works with `--no-config --vo=opengl` too and in this case (no ICC profile) the refresh rate should be known at a pretty similar time for both back-ends?

**Comment by wm4** [comment]
Actually, x11_common returns the FPS of the first display as fallback if none is known. I guess this was done for a similar case (needing to know display fps before video init, or so).

**Comment by Argon-** [comment]
But this shouldn't result in 0, should it? Maybe the Hz of the wrong display, but 0 in the VS script indicates no returned value for the `VOCTRL_GET_DISPLAY_FPS` request (imo).

**Comment by wm4** [comment]
Right, I missed another check which actually makes VOCTRL_GET_DISPLAY_FPS return which failure if no window was mapped yet.

**Comment by Argon-** [comment]
I compiled mpv with x11 backend and it's not working with x11 here too. The first call to `vo_x11_update_geometry` is too late, the cocoa backend seems to query `vo_cocoa_update_screen_info` earlier/early enough.

Is there something one can do about this?

**Comment by wm4** [comment]
Either x11 could return the refresh rate of a random monitor (wouldn't be too bad; it appears vsync with multiple monitors is pretty broken with most drivers), or pick a different approach.

**Comment by wm4** [comment]
Anyway, not like it matters; changed x11 to the first suggestion and merged.

**Comment by haasn** [comment]
I've tested it on the nvidia drivers, results:

1. Argon's patch still gets me 0.0
2. wm4's patch gets me the refresh rate of the wrong display (secondary monitor, rather then primary)
